types/mocha/index.d.ts

```diff
@@ -5,10 +5,28 @@
 //                 jt000 <https://github.com/jt000>
 //                 Vadim Macagon <https://github.com/enlight>
 //                 Andrew Bradley <https://github.com/cspotcode>
+//                 Dmitrii Sorin <https://github.com/1999>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 // TypeScript Version: 2.1
 
-interface MochaSetupOptions {
+/**
+ * Mocha is using node.js EventEmitter for its internal classes
+ * But when it's executed in the browser environment it bundles a 3rd party EventEmitter in its code
+ * Also node.js EventEmitter brings its globals which cannot be used in the browser environment.
+ * So it's unsafe to reference node EventEmitter here, that's why we have a generic interface for EventEmitter.
+ *
+ * @see https://github.com/DefinitelyTyped/DefinitelyTyped/pull/25117#issuecomment-383404187
+ */
+declare class GenericEventEmitter {
+    addListener(event: string, listener: (...args: any[]) => void): GenericEventEmitter;
+    on(event: string, listener: (...args: any[]) => void): GenericEventEmitter;
+    once(event: string, listener: (...args: any[]) => void): GenericEventEmitter;
+    removeListener(event: string, listener: (...args: any[]) => void): GenericEventEmitter;
+    removeAllListeners(event?: string): GenericEventEmitter;
+    emit(event: string, ...args: any[]): boolean;
+  }
```

Comment:
@Flarna after our discussion I decided to include a small interface for environment-agnostic EventEmitter right into the typings. Hopefully it's okay 🤞